src/component/axis/AxisBuilder.ts

```diff
@@ -420,10 +420,26 @@ const builders: Record<'axisLine' | 'axisTickLabel' | 'axisName', AxisElementsBu
 
         const textFont = textStyleModel.getFont();
 
+        const coordSysModel = axisModel.getCoordSysModel();
+        const coordSys = coordSysModel ? coordSysModel.coordinateSystem : null;
+
         const truncateOpt = axisModel.get('nameTruncate', true) || {};
-        const ellipsis = truncateOpt.ellipsis;
+        const ellipsis = textStyleModel.get('ellipsis') || truncateOpt.ellipsis;
+        const overflow = textStyleModel.get('overflow') || 'none';
+
+        let nameTruncateDefaultWidth;
+
+        if (overflow !== 'none') {
+            nameTruncateDefaultWidth = Math.abs(extent[1] - extent[0]); // Current axis length.
+            if (!isNameLocationCenter(nameLocation)) {
+                const rect = coordSys && coordSys.getRect && coordSys.getRect();
+                nameTruncateDefaultWidth = rect ? rect.width : nameTruncateDefaultWidth * 2;
```

Comment:
It's odd to use `nameTruncateDefaultWidth * 2` because is this is assuming we are using polar coordinate. @pissang @100pah Please help further review this PR.